src/Microsoft.Health.Fhir.Core/Features/Conformance/ConformanceHealthCheck.cs

```diff
@@ -0,0 +1,73 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Health.Fhir.Core.Features.Conformance
+{
+    public class ConformanceHealthCheck : IHealthCheck
+    {
+        private readonly IConformanceProvider _systemConformanceProvider;
+        private readonly ILogger<ConformanceHealthCheck> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CosmosHealthCheck"/> class.
+        /// </summary>
+        /// <param name="systemConformanceProvider">The provider for the capability statement/</param>
+        /// <param name="logger">The logger.</param>
+        public ConformanceHealthCheck(
+            IConformanceProvider systemConformanceProvider,
+            ILogger<ConformanceHealthCheck> logger)
+        {
+            EnsureArg.IsNotNull(systemConformanceProvider, nameof(systemConformanceProvider));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _systemConformanceProvider = systemConformanceProvider;
+            _logger = logger;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var metadata = await _systemConformanceProvider.GetMetadata(cancellationToken);
+
+                if (metadata != null)
+                {
+                    return HealthCheckResult.Healthy("Successfully retrieved the capability statement.");
+                }
+                else
+                {
+                    _logger.LogWarning("Capability statement was null.");
+
+                    return HealthCheckResult.Unhealthy("Failed to retrieve the capability statement.");
+                }
+            }
+            catch (ArgumentNullException ex)
+            {
+                if (ex.StackTrace.Contains("UrlHelperFactory", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    // If the metadata document hasn't been initialized yet before health check is called the Url Helper factory will not have an action context from this call. The system is still healthy.
```

Comment:
Url Helper factory doesn't have an ActionContext I think because /health/check isn't an MVC route

Reply:
I think adding a MVC route to /health/check will be a more complex task. 
I've noticed how we are doing that for /metadata in FhirController, but that's something we should test more carefully.

Reply:
I don't think we should add a MVC route for health/check. Reading the documentation on health check endpoints, they are meant to have minimal dependencies and operate quickly. I think having it just check the cached metadata is sufficient for health/check, it shouldn't be trying to construct the metadata document. The special case we have here allows it to detect if the metadata document can't be constructed, but still allows the health check to be fast and reliable in a healthy system.

Reply:
I think we should spend more time finding a solution to this problem.
I don't like the check if the exception came from _UrlHelperFactory_, just because if another dependency to _ActionContext_ raises the _ArgumentNullException_ it'll break the /health/check.